GameServerLib/API/ApiFunctionManager.cs

```diff
@@ -116,11 +116,26 @@ public static bool IsWalkable(float x, float y)
             return _game.Map.NavGrid.IsWalkable(x, y);
         }
 
-        public static void AddBuff(string buffName, float duration, byte stacks, BuffType buffType, IObjAiBase onto, IObjAiBase from)
+        public static void AddBuff(string buffName, float duration, byte stacks, BuffType buffType, IObjAiBase onto, IObjAiBase from, float removeAfter = -1.0f)
         {
+            if (duration < 0)
+            {
+                return;
+            }
+
             var buff = new Buff(_game, buffName, duration, stacks, buffType, onto, from);
+
             onto.AddBuff(buff);
```

Comment:
If you are fine with changing the title of this PR to something like "Expanding Scripting API: AddBuff", you could add to this function the ability to call AddBuffGameScript.
You will need to do:
`public static void AddBuff(string buffName, float duration, byte stacks, BuffType buffType, ISpell spell, IObjAiBase onto, IObjAiBase from, float removeAfter = -1.0f, bool isUnique = false)`
then after `onto.AddBuff(buff);` you will need to add `onto.AddBuffGameScript(buffName, buffName, spell, removeAfter, isUnique);`

Reply:
Could also probably remove all of `removeAfter` and replace with `duration`.